crates/cxx-qt-gen/src/generator/cpp/property/mod.rs

```diff
@@ -27,24 +27,49 @@ pub fn generate_cpp_properties(
     let qobject_ident = qobject_idents.name.cxx_unqualified();
 
     for property in properties {
-        // Cache the idents as they are used in multiple places
+        // Cache the idents and flags as they are used in multiple places
         let idents = QPropertyNames::from(property);
         let cxx_ty = syn_type_to_cpp_type(&property.ty, type_names)?;
 
+        let flags = &property.flags;
+
         generated.metaobjects.push(meta::generate(&idents, &cxx_ty));
-        generated
-            .methods
-            .push(getter::generate(&idents, &qobject_ident, &cxx_ty));
-        generated
-            .private_methods
-            .push(getter::generate_wrapper(&idents, &cxx_ty));
-        generated
-            .methods
-            .push(setter::generate(&idents, &qobject_ident, &cxx_ty));
-        generated
-            .private_methods
-            .push(setter::generate_wrapper(&idents, &cxx_ty));
-        signals.push(signal::generate(&idents, qobject_idents));
+
+        // None indicates no custom identifier was provided
+        if flags.read.is_none() {
+            generated
+                .methods
+                .push(getter::generate(&idents, &qobject_ident, &cxx_ty));
+            generated
+                .private_methods
+                .push(getter::generate_wrapper(&idents, &cxx_ty));
+        }
+
+        // Checking custom setter wasn't provided
+        if flags
+            .write
+            .clone()
```

Comment:
i think instead of `.clone()` you can use `.as_ref()` here, but this would also go away if you go the enum idea route

Reply:
done, this was changed